Framework/Kernel/src/AttenuationProfile.cpp

```diff
@@ -55,13 +64,33 @@ AttenuationProfile::AttenuationProfile(const std::string &inputFileName,
     std::ifstream input(inputFilePath.toString(), std::ios_base::in);
     if (input) {
       std::string line;
+      double minLambda = std::numeric_limits<double>::max();
+      double maxLambda = std::numeric_limits<double>::min();
       while (std::getline(input, line)) {
         double lambda, alpha, error;
         if (std::stringstream(line) >> lambda >> alpha >> error) {
```

Comment:
Just to confirm, is this line checking if you can get at least 3 values out of the line? Is this the best way to test this? what happens if the line contains only two values?

Reply:
I may well have written this previously...
The code "if (stringstream)" checks whether the stringstream is happy or not.
Each action of the >> operator returns the stringstream itself in addition to writing some text into a variable (lambda, alpha etc). So the if statement in the code above is evaluating the stringstream state after attempts to read 3 strings from it.
If there are only two values on a line in the input file, the if statement returns false and it just moves onto the next iteration of the while loop and reads the next line.
The .DAT file format that we've been asked to support contains some header lines where there aren't three columns so this behaviour kicks in while scanning past those lines (and ignoring them). So I think this works OK in this situation - possible elsewhere we may prefer to raise an error but ignoring is OK here.

Reply:
What is the result if there are 4 componants to `line`?

Reply:
I'm pretty sure it would just read the first 3 values and ignore the 4th. 
One more point about the validation that is done by this line. The values read in have to match the data types of the variables for the if statement to evaluate to true. So if a line contained 3 strings then the if(stringstream) would evaluate to false and the line would be ignored.
The data file that I've been given for this is DIAM_DEBEERS_MU.DAT which you should see in the instrument\sampleenvironments folder.